tesseract_motion_planners/descartes/include/tesseract_motion_planners/descartes/impl/descartes_robot_sampler.hpp

```diff
@@ -99,40 +99,44 @@ std::vector<descartes_light::StateSample<FloatType>> DescartesRobotSampler<Float
       }
       else
       {
-        const FloatType cost = static_cast<FloatType>(collision_->distance(sol));
-        samples.push_back(descartes_light::StateSample<FloatType>{ state, cost });
+        if (!allow_collision_)
+        {
+          if (collision_->validate(sol))
+            samples.push_back(descartes_light::StateSample<FloatType>{ state, 0.0 });
+        }
```

Comment:
This can be moved to just be an `else if` rather than another nested if statement inside the outer else, that way it will be easier to read.

Reply:
Done. @marrts good catch.